src/StatsParent.js

```diff
@@ -13,51 +13,37 @@ class StatsParent {
   }
 
   //used in sleep, hydration
-  returnWeek(numWeeks) {
-    var specificUser = this.findUser()
+  returnWeek(numWeeks, userData, id) {
+    var specificUser = this.findUser(userData, id)
     return [...specificUser].splice(-7 * numWeeks, 7).map(day => day.date);
   }
-
+  //used in sleepREpo, ActivityREpo
+  getAverageFromDataList(dataList, relevantProperty) {
+    return Number((dataList.reduce((totalQuality, eachPerson) => {
+      totalQuality += eachPerson[relevantProperty];
+      return totalQuality;
+    }, 0) / dataList.length).toFixed(2));
+  }
   //used in sleep, hydration
-  returnAvgData(relevantProperty) {
-      let specificUser = this.findUser();
-      return Number((specificUser.reduce((totalHours, daily) => {
-        totalHours += daily[relevantProperty];
-        return totalHours;
-      }, 0) / specificUser.length).toFixed(2));
+  returnAvgData(userData, id, relevantProperty) {
+      let specificUser = this.findUser(userData, id);
+      return this.getAverageFromDataList(specificUser, relevantProperty)
     }
     //used in sleep, activity, hydration
-  returnUserDataForDay(date, relevantProperty) {
-    let specificUser = this.findUser();
+  returnUserDataForDay(userData, id, date, relevantProperty) {
+    let specificUser = this.findUser(userData, id);
     return specificUser.find(day => day.date === date)[relevantProperty];
   }
   //used in sleep, hydration
-  returnWeekOfSpecificData(numWeeks, relevantProperty) {
-    let specificUser = this.findUser();
+  returnWeekOfSpecificData(numWeeks, userData, id, relevantProperty) {
```

Comment:
From just reading the method name here I assume we are trying to get one week of specific data but parameter is written as numWeeks I think it would be more clear to say weekNum so it's more clear that we are looking for that specific week.

Reply:
Or `weekNumber`! `specificWeek` could also be good. Keep those params specific to what their purpose is